internal/provider/kubernetes/controller.go

```diff
@@ -495,6 +495,14 @@ func (r *gatewayAPIReconciler) statusUpdateForGateway(gtw *gwapiv1b1.Gateway, sv
 			}
 			gCopy := g.DeepCopy()
 			gCopy.Status.Conditions = status.MergeConditions(gCopy.Status.Conditions, gtw.Status.Conditions...)
+			for index := range gCopy.Status.Conditions {
```

Comment:
thanks for fixing the issue @Xunzhuo !
I have two questions
* why are we updating all the conditions here ?
isn't this already done correctly in https://github.com/envoyproxy/gateway/blob/085e6fa5a56066472a52e8c07bc1aefa3ffca5a2/internal/status/conditions.go#L116
* shouldn't this logic be living in https://github.com/envoyproxy/gateway/blob/main/internal/status/conditions.go
cc @skriss  who was also looking into this

Reply:
please ignore this in favor of https://github.com/envoyproxy/gateway/pull/853#issuecomment-1371423935